### PR TITLE
Make hover spikes work when no tick labels are present

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/1977 and makes https://github.com/plotly/plotly.js/pull/1604 obsolete.

@alexcjohnson I wouldn't mind a +1 on my approach before I add a test or two.

### CODE REVIEW

**Comment by alexcjohnson** [comment]
Looks solid to me. Might want to calculate the rest of the bbox attributes just so people can pretend it's a real bbox (I see one caller that looks for `.height` for example).

File: test/jasmine/tests/hover_spikeline_test.js [hunk 1]
```diff
@@ -34,6 +38,20 @@ describe('spikeline', function() {
...
 
+        it('draws lines and markers on enabled axes w/o tick labels', function(done) {
+            Plotly.relayout(gd, {
+                'xaxis.showticklabels': false,
+                'yaxis.showticklabels': false
+            })
+            .then(function() {
+                Fx.hover('graph', {xval: 2, yval: 3}, 'xy');
+                expect(d3.selectAll('line.spikeline').size()).toEqual(4);
+                expect(d3.selectAll('circle.spikeline').size()).toEqual(1);
```
Comment (alexcjohnson) [inline test/jasmine/tests/hover_spikeline_test.js]: is it worth verifying that these things have reasonable dimensions/positions?

Comment (etpinard) [reply test/jasmine/tests/hover_spikeline_test.js]: Good call. done in 92b606b

**Comment by alexcjohnson** [comment]
Looks fantastic - thanks for testing positions. 💃

**Pull request merged**